### PR TITLE
Add validation warning when filter is applied to conversion input measure

### PR DESCRIPTION
**Pull request by WilliamDee**
Resolves SL-2862

### Description
Since filtering on a conversion input measure is not fully supported, we will add a warning when this is added to let the user know.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: dbt_semantic_interfaces/validations/metrics.py [hunk 1]
```diff
@@ -519,17 +522,31 @@ def _validate_measure(measure_reference: MeasureReference, semantic_model: Seman
                     )
                 )
 
+            if input_measure.filter is not None and not is_base_measure:
+                # Filters for conversion measure input is not fully supported.
```
Comment (courtneyholcomb) [inline dbt_semantic_interfaces/validations/metrics.py:526]: nit: are*

File: dbt_semantic_interfaces/validations/metrics.py [hunk 2]
```diff
@@ -519,17 +522,31 @@ def _validate_measure(measure_reference: MeasureReference, semantic_model: Seman
...
+                issues.append(
+                    ValidationWarning(
+                        context=MetricContext(
+                            file_context=FileContext.from_metadata(metadata=metric.metadata),
+                            metric=MetricModelReference(metric_name=metric.name),
+                        ),
+                        message=f"For conversion metrics, Measure input {measure.name} has a filter,"
+                        " filter on a conversion input measure is not fully supported.",
```
Comment (courtneyholcomb) [inline dbt_semantic_interfaces/validations/metrics.py:534]: maybe let's say "yet"?

**Review by courtneyholcomb — APPROVED** [review]
That was so fast 😂 
Thank you also for the cleanup!!
Note that this will need a version bump (prod version) so we can release it and get it into core. And then a core PR that bumps to that version.

**Comment by WilliamDee** [comment]
> Note that this will need a version bump (prod version) so we can release it and get it into core. And then a core PR that bumps to that version.

@courtneyholcomb Are there any other validation tasks? Would it make sense to wait for some of those to be done then we can batch all those up and release it once?

**Comment by courtneyholcomb** [comment]
> > Note that this will need a version bump (prod version) so we can release it and get it into core. And then a core PR that bumps to that version.
> 
> @courtneyholcomb Are there any other validation tasks? Would it make sense to wait for some of those to be done then we can batch all those up and release it once?

@WilliamDee 
Good call, yeah I think there are! Can you put up a task to make sure we don't forget?

**Pull request merged**